SwrveSDK/src/main/java/com/swrve/sdk/SwrveImp.java

```diff
@@ -645,7 +651,7 @@ protected void findCacheFolder(Activity activity) {
     }
 
     protected boolean checkPermissionGranted(Context context, String permission) {
-        return ActivityCompat.checkSelfPermission(context, permission) == PackageManager.PERMISSION_GRANTED;
+        return SwrveHelper.checkPermissionGranted(context, permission);
```

Comment:
Probably best to just call SwrveHelper.checkPermissionGranted directly instead of going through this method. Also, might aswell move requestPermissions helper to the SwrveHelper class too and keep them together.